homeassistant/components/awair/config_flow.py

```diff
@@ -21,20 +23,79 @@ class AwairFlowHandler(ConfigFlow, domain=DOMAIN):
 
     VERSION = 1
 
+    _device: AwairLocalDevice
+
+    async def async_step_zeroconf(
+        self, discovery_info: zeroconf.ZeroconfServiceInfo
+    ) -> FlowResult:
+        """Handle zeroconf discovery."""
+
+        host = discovery_info.host
+        LOGGER.debug("Discovered device: %s", host)
+
+        self._device, _ = await self._check_local_connection(host)
+
+        if self._device is not None:
+            await self.async_set_unique_id(self._device.mac_address)
+            self._abort_if_unique_id_configured(updates={CONF_HOST: host})
+            self.context.update(
+                {
+                    "title_placeholders": {
+                        "host": host,
+                        "model": self._device.model,
+                    }
+                }
+            )
+        else:
+            return self.async_abort(reason="unreachable")
+        return await self.async_step_discovery_confirm()
+
+    async def async_step_discovery_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Confirm discovery."""
+        if user_input is not None:
+            title = f"{self._device.model} ({self._device.device_addr})"
```

Comment:
The address can change if the IP address changes. So let's not put it in the name.

Can we fetch the name of the Awair device if the user has given it one?

Reply:
Name isn't returned in the local API. Here is a dump of whats available locally:

```json
{
  "device_uuid":"awair-element_12345",
  "wifi_mac":"REDACTED",
  "ssid":"Internet of Shit",
  "ip":"10.0.5.188",
  "netmask":"255.255.255.0",
  "gateway":"none",
  "fw_version":"1.2.8",
  "timezone":"America/Los_Angeles",
  "display":"score",
  "led":{
    "mode":"auto",
    "brightness":179
  },
  "voc_feature_set":34
}
```

Reply:
I've removed the host for now. I think there should be some unique identifier shown so that the user knows which device they're setting up, but its not clear what the right choice is here.

Reply:
I think I will do `f"{self._device.model} ({self._device.device_id})"`

`device_id` is the last 5 digits on `device_uuid`